### PR TITLE
[azure] delete role assignment on bastion host

### PR DESCRIPTION
**Pull request by jinyunma**
when installing cluster in bastion host, we have two roles assigned on bastion host in the scope of subscription level, but forgot to delete those role assignment in deprovision stage.

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-azure-ipi-oidc-managed-identity-system-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-azure-ipi-oidc-managed-identity-system-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-azure-ipi-oidc-managed-identity-system-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.17-azure-ipi-oidc-managed-identity-system-f14

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei to review

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/deprovision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-deprovision-chain.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ chain:
   - ref: gather-azure-resource
   - chain: gather
   - ref: ipi-deprovision-deprovision-in-bastion
+  - ref: azure-deprovision-role-assignment
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/deprovision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-deprovision-chain.yaml:9]: Shall we rename it to `azure-deprovision-bastionhost-role-assignment`

Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/deprovision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-deprovision-chain.yaml:9]: hmm, this step `azure-deprovision-role-assignment` is originally intended to remove any role assignment if file `"${SHARED_DIR}/azure_role_assignment_ids` exists, not only for bastion host.
If there is any identity attached on other resources assigns role on scope of subscription in the future, we only need to append role assignment id to file `azure_role_assignment_ids` and this step can be reused.

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/cco-manual-workload-identity/managed-identity/deprovision/cucushift-installer-rehearse-azure-ipi-cco-manual-workload-identity-managed-identity-deprovision-chain.yaml:9]: Got it.

**Pull request merged**